### PR TITLE
Nalu: updating Trilinos recipe a bit

### PR DESCRIPTION
**Pull request by lucbv**
Basic changes to build/install nalu properly using Spack. Some more changes would be nice for instance adding an option to build against Trilinos master or develop. Adding a dependency on googletest to avoid the annoying build failures in the unit-tests.

### CODE REVIEW

File: var/spack/repos/builtin/packages/nalu/package.py [hunk 1]
```diff
@@ -31,21 +31,21 @@ class Nalu(CMakePackage):
...
-    depends_on("yaml-cpp@0.5.3:", when="+shared")
-    depends_on("yaml-cpp~shared@0.5.3:", when="~shared")
+    depends_on("yaml-cpp@0.5.3:0.6.2", when="+shared")
+    depends_on("yaml-cpp~shared@0.5.3:0.6.2", when="~shared")
     # Cannot build Trilinos as a shared library with STK on Darwin
     # which is why we have a 'shared' variant for Nalu
     # https://github.com/trilinos/Trilinos/issues/2994
     depends_on(
-        "trilinos"
+        "trilinos@develop"
```
Comment (psakievich) [inline var/spack/repos/builtin/packages/nalu/package.py]:
```suggestion
        "trilinos@master:"
```
You can use the versioning semantics to allow for multiple named versions. See bullet 1 in 
https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

**Comment by lucbv** [comment]
I am going to avoid using develop, see my new changes, in favor of adding a released version that is verifiably building and the master branch will just depend on the state of the repo... nothing can really be done to ensure that there is no incompatibilities there.

I am adding the `+gtest` variant to Trilinos which allows nalu to build its unit-tests, `gotype=long` is just a strong requirement, not sure why/how things use to build without it? Finally the yaml-cpp version are constrained as Nalu is not building against more recent versions of the library.

@psakievich if you can have one more look at it that would be nice, assuming you are fine with this we can ask a spack member to review as well

**Comment by lucbv** [comment]
@tldahlgren do you mind having a look at this PR.
It mostly cleans things up and provides an actual recipe for a versioned release of Nalu!

**Review by tldahlgren — APPROVED** [review]
Confirmed the new version sha256, rest LGTM.

@psakievich Do you have any more feedback on this PR?

File: var/spack/repos/builtin/packages/nalu/package.py [hunk 2]
```diff
@@ -15,9 +15,14 @@ class Nalu(CMakePackage):
     """
 
     homepage = "https://github.com/NaluCFD/Nalu"
+    url = "https://github.com/NaluCFD/Nalu/archive/refs/tags/v1.6.0.tar.gz"
     git = "https://github.com/NaluCFD/Nalu.git"
 
     version("master", branch="master")
+    version("1.6.0", sha256="2eafafe25ed44a7bc1429881f8f944b9794ca51b1e1b29c28a45b91520c7cf97")
```
Comment (psakievich) [inline var/spack/repos/builtin/packages/nalu/package.py:22]: @lucbv  if you still want to have the option to build against `trilinos@develop` you could do something like:
``` python
verison("master", branch="master", preferred=True)
version("develop", branch="master")
depends_on("trilinos@develop", when="@=develop")
```

**Review by psakievich — APPROVED** [review]
This LGTM.  A comment for @lucbv but it really depends on his desired outcome for the package.

File: var/spack/repos/builtin/packages/nalu/package.py (hunk 2 above)
Comment (lucbv) [reply var/spack/repos/builtin/packages/nalu/package.py:22]: Yes, I saw the `preferred` argument but I'm okay with not adding develop, that's probably better that way as I believe Nalu runs its CI against Trilinos master but not develop. If you all are fine with this I think we can go ahead and merge

**Pull request merged**